include/ssd-internal.h

```diff
@@ -117,16 +156,23 @@ struct ssd {
 	struct border margin;
 };
 
+/*
+ * ssd_part wraps a scene-node with ssd-specific information and can be
+ * accessed with node_ssd_part_from_node(wlr_scene_node *).
+ * This allows get_cursor_context() in desktop.c to see which SSD part is under
+ * the cursor.
+ */
 struct ssd_part {
 	enum ssd_part_type type;
-
-	/* Buffer pointer. May be NULL */
-	struct scaled_font_buffer *buffer;
+	struct view *view;
 
 	/* This part represented in scene graph */
 	struct wlr_scene_node *node;
+	struct wl_listener node_destroy;
 
-	struct wl_list link;
+	/* Set when type == LAB_SSD_BUTTON_* */
+	struct ssd_button *button;
+	struct wl_list link; /* ssd_titlebar_subtree.buttons_{left,right} */
 };
```

Comment:
This is `struct ssd_part`, should we omit these two members and instead create a new struct like
```c
struct ssd_part_button {
	struct ssd_part base;
	struct ssd_button *button;
	struct wl_list link;
}
```
Or as alternative include `struct ssd_part base` as first entry in `struct ssd_part`?
These approaches would prevent having to add optional elements to `struct ssd_part` for different parts.

I don't feel strongly about it though.

Reply:
Looks good